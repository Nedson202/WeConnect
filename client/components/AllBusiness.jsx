import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Navbar from './Navbar.jsx';
import BusinessList from './BusinessList.jsx';
import Footer from './Footer.jsx';
import { fetchBusinesses } from '../actions/fetchBusinessAction';

class AllBusiness extends Component {
  constructor(props){
    super(props);

    this.state = {
      isLoading: false
    };
  }

  componentWillMount() {
    document.title = 'Business list'
    this.setState({isLoading: true})

    this.props.fetchBusinesses().then(
      () => {
        this.setState({isLoading: false})
      }
    );
  }

  render() {
    const { businesses } = this.props;
    const { isLoading } = this.state;

    if(isLoading) {
      return (
        <div>
          <Navbar />

          <div id="loader"></div>

          <Footer />
        </div>
      );
    }

    return (
      <div>
        <Navbar />

        <BusinessList businesses={businesses}/>

        <Footer />
      </div>
    );
  }
}

AllBusiness.propTypes = {
  businesses: PropTypes.array.isRequired,
  fetchBusinesses: PropTypes.func.isRequired,
}

function mapStateToProps(state) {
  return {
    businesses: state.businesses
  }
}

export default connect(mapStateToProps, { fetchBusinesses })(AllBusiness);

// import React, { Component } from 'react';
// import PropTypes from 'prop-types';
// import { connect } from 'react-redux';
// import Navbar from './Navbar.jsx';
// import BusinessList from './BusinessList.jsx';
// import Footer from './Footer.jsx';
// import { fetchBusinesses } from '../actions/fetchBusinessAction';
//
// class AllBusiness extends Component {
//   constructor(props){
//     super(props);
//
//     this.state = {
//       isLoading: false
//     };
//   }
//
//   componentWillMount() {
//     document.title = 'Business list'
//     this.setState({isLoading: true})
//
//     this.props.fetchBusinesses().then(
//       () => {
//         this.setState({isLoading: false})
//       }
//     );
//   }
//
//   render() {
//     const { businesses } = this.props;
//     const { isLoading } = this.state;
//
//     if(isLoading) {
//       return (
//         <div>
//           <Navbar />
//
//           <div id="loader"></div>
//
//           <Footer />
//         </div>
//       );
//     }
//
//     return (
//       <div>
//         <Navbar />
//
//         <BusinessList businesses={businesses}/>
//
//         <Footer />
//       </div>
//     );
//   }
// }
//
// AllBusiness.propTypes = {
//   businesses: PropTypes.array.isRequired,
//   fetchBusinesses: PropTypes.func.isRequired,
// }
//
// function mapStateToProps(state) {
//   return {
//     businesses: state.businesses
//   }
// }
//
// export default connect(mapStateToProps, { fetchBusinesses })(AllBusiness);
