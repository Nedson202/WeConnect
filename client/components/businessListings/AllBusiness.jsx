import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import BusinessList from '../common/BusinessList';
import { fetchBusinesses } from '../../actions/fetchActions';
import Spinner from '../common/Spinner';
import loader from '../../actions/loader';
/**
 * @class AllBusiness
 *
 * @extends {Component}
 */
class AllBusiness extends Component {
  /**
   * @description Fetch reviews and business
   *
   * @returns {undefined}
   *
   * @memberof AllBusiness
   */
  componentWillMount() {
    this.props.loader(true);
    document.title = 'Business list';
    this.props.fetchBusinesses('page=1');
  }
  /**
   * @description Renders the component to the dom
   *
   * @returns {object} JSX object
   *
   * @memberof BusinessProfile
   */
  render() {
    const { businesses, paginate, isLoading } = this.props;

    const allBusiness = businesses || [];

    if (isLoading) {
      return <Spinner isLoading={isLoading} />;
    }

    return (
      <div>
        { allBusiness.length === 0 ? null :
        <h2 style={{ paddingTop: '70px' }} className="text-center">Business Listings</h2>}
        <BusinessList
          businesses={allBusiness}
          paginate={paginate}
          fetchBusinesses={this.props.fetchBusinesses}
        />
      </div>
    );
  }
}

AllBusiness.propTypes = {
  fetchBusinesses: PropTypes.func.isRequired,
  loader: PropTypes.func.isRequired,
  businesses: PropTypes.array.isRequired,
  paginate: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => {
  const { businesses, isLoading } = state;
  return {
    businesses: businesses.businesses,
    paginate: businesses.paginationResult,
    isLoading
  };
};

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchBusinesses,
  loader
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AllBusiness);
