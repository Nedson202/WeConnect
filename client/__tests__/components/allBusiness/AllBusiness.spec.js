import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';import toJSON from 'enzyme-to-json';
import connectedAllBusiness, { AllBusiness} 
from '../../../components/businessListings/AllBusiness';
import businessData from '../../__mockData__/business';
import { mapStateToProps } from '../../../components/businessListings/AllBusiness';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const { allBusiness, paginationResult } = businessData;
configure({ adapter: new Adapter() });

let props, businesses;
const setup = () => {
  props = {
    fetchBusinesses: jest.fn(() => Promise.resolve()),
    businesses: allBusiness,
    paginate: paginationResult,
    isLoading: false,
    loader: jest.fn(() => Promise.resolve())
  };

  return shallow(<AllBusiness {...props} />);
};


describe('Component: AllBusiness', () => {
  beforeEach(() => {
    global.document = {
      title: () => {}
    }
  })

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('Spinner').length).toBe(0);
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('h2').length).toBe(1);
    expect(wrapper.find('BusinessList').length).toBe(1);
    wrapper.setProps({ isLoading: true, businesses: [] });
    expect(wrapper.find('Spinner').length).toBe(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.length).toBe(1);

    const state = {
      businesses: {
        businesses: allBusiness
      }
    }
    expect(mapStateToProps(state).businesses).toEqual(allBusiness);
    wrapper.setProps({ isLoading: true, businesses: [] });
    expect(wrapper.find('h2').length).toBe(0);
  })
});

describe('state to props', () => {
  it('should render connected allbusinesses page', () => {
    const businesses = {
      businesses,
      paginationResult
    };

    const store = mockStore({
      businesses,
      isLoading: false
    });
    const wrapper = shallow(<connectedAllBusiness store={store}  />);
    expect(wrapper.length).toBe(1);
  });
});