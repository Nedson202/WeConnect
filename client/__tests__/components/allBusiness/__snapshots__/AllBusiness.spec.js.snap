// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AllBusiness should render 1`] = `
<Spinner
  isLoading={true}
/>
`;
